zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/RoleEntityAddedApplier.java

```diff
@@ -31,11 +30,14 @@ public RoleEntityAddedApplier(final MutableProcessingState state) {
   @Override
   public void applyState(final long key, final RoleRecord value) {
     roleState.addEntity(value);
-    if (EntityType.USER.equals(value.getEntityType())) {
-      userState.addRole(value.getEntityKey(), value.getRoleKey());
-    }
-    if (EntityType.MAPPING.equals(value.getEntityType())) {
-      mappingState.addRole(value.getEntityKey(), value.getRoleKey());
+    switch (value.getEntityType()) {
+      case USER -> userState.addRole(value.getEntityKey(), value.getRoleKey());
+      case MAPPING -> mappingState.addRole(value.getEntityKey(), value.getRoleKey());
+      default ->
+          throw new IllegalStateException(
+              "Unknown or unsupported entity type: '"
+                  + value.getEntityType()
+                  + "'. Please contact support for clarification.");
```

Comment:
🔧 Could you change this so that we follow our [error message guidelines](https://github.com/camunda/camunda/wiki/Error-Guidelines)?

Eg. `Expected to add entity '<key>' to role '<key>', but entities of type '<type>' cannot be added to roles`